### PR TITLE
Update 4.X branch to benefit latest releases

### PR DESCRIPTION
**Pull request by ZacharieTFR**
### Description
The last releases [3.1.1](https://github.com/dailymotion/vast-client-js/releases/tag/3.1.1) and [3.1.0](https://github.com/dailymotion/vast-client-js/releases/tag/3.1.0) introduced new features and fixes that are missing in the 4.X branch. 
Rebased 4.X to master and resolved conflicts.

### Type
- [ ] Breaking change
- [X] Enhancement
- [ ] Fix
- [ ] Documentation
- [ ] Tooling

### CODE REVIEW

File: src/vast_tracker.js [hunk 1]
```diff
@@ -597,46 +605,51 @@ export class VASTTracker extends EventEmitter {
...
       }
       if (this.progress) {
-        macros['ADPLAYHEAD'] = this.progressFormatted();
+        givenMacros['ADPLAYHEAD'] = this.progressFormatted();
       }
     }
     if (this.creative?.universalAdIds?.length) {
-      macros['UNIVERSALADID'] = this.creative.universalAdIds
-        .map(universalAdId => universalAdId.idRegistry.concat(' ', universalAdId.value))
-        .join(',');
```
Comment (clementFrancon) [inline src/vast_tracker.js:616]: Why removing the join here ? 
I see the specs saying "Multiple UniversalAdIDs can be supported with a comma separating them."

**Review by clementFrancon — COMMENTED** [review]
Small question otherwise LGTM

File: src/vast_tracker.js (hunk 1 above)
Comment (ZacharieTFR) [reply src/vast_tracker.js:616]: Good catch! 👍 It was a fail during the merge conflict resolution. Resolved in https://github.com/dailymotion/vast-client-js/pull/382/commits/caea3e9dd63f5e78fe8a678314b1ff50d544ffcf

**Review by clementFrancon — APPROVED** [review]

**Review by rumesh — APPROVED** [review]
🚢

**Pull request merged**